app/routes/admin.py

```
@@ -4326,22 +4327,25 @@ def student_detail_public(actor_public_id):
+        _entitlement_query = _entitlement_query.filter(EntitlementEvent.class_id == class_id)
+    _entitlements_raw = _entitlement_query.order_by(EntitlementEvent.timestamp.desc()).all()
     store_purchases = [
         SimpleNamespace(
             id=ent.entitlement_id,
             seat_id=ent.target_seat_id,
             class_id=ent.class_id,
-            store_item=db.session.get(StoreItem, ent.entitlement_item_id),
-            store_item_id=ent.entitlement_item_id,
+            store_item=db.session.get(StoreItem, ent.product_id),
+            store_item_id=ent.product_id,
             status=derive_display_status(ent.entitlement_id),
```

Ruff (0.16.0)

[error] Undefined name `derive_display_status`

(F821)